### PR TITLE
test: adding java 7 check

### PR DESCRIPTION
**Pull request by suztomo**
We haven't run the Java 7 builds for a while even though the README.md says Java 7 target. Let's clarify the current situation by installing the check again.

It turned out that the google-http-client-jackson2 and google-http-client-appengine modules require Java 8 or higher due to their dependencies. We keep the compilation target for Java 7 so that the users can tweak some old version of the dependencies.

The surefire (unit test) plugin's ["jvm" option](https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#jvm) is the core of this Java 7 check. Guava's warning about Java 7 tells that the test is indeed running on Java 7:

image

https://github.com/googleapis/google-http-java-client/actions/runs/4927699269/jobs/8805047743?pr=1847

Fixes https://github.com/googleapis/google-http-java-client/issues/1713
Closes https://github.com/googleapis/google-http-java-client/pull/1721

### CODE REVIEW

**Comment by suztomo** [comment]
It turned out that Maven wrapper does not work with Java 7.

```
Error: Exception in thread "main" java.lang.UnsupportedClassVersionError: org/apache/maven/wrapper/MavenWrapperMain : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:808)
```

**Comment by suztomo** [comment]
Memo: Maven 3.5.x does not have "-ntp" option.

**Comment by suztomo** [comment]
3.5.4 still failed:

```
constituent[0]: file:/opt/hostedtoolcache/maven/3.5.4/x64/conf/logging/
constituent[1]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-plugin-api-3.5.4.jar
constituent[2]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/plexus-interpolation-1.24.jar
constituent[3]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-resolver-util-1.1.1.jar
constituent[4]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-resolver-impl-1.1.1.jar
constituent[5]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/plexus-utils-3.1.0.jar
constituent[6]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/jsr250-api-1.0.jar
constituent[7]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/commons-cli-1.4.jar
constituent[8]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/jcl-over-slf4j-1.7.25.jar
constituent[9]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-compat-3.5.4.jar
constituent[10]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-artifact-3.5.4.jar
constituent[11]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-settings-builder-3.5.4.jar
constituent[12]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-repository-metadata-3.5.4.jar
constituent[13]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/guice-4.2.0-no_aop.jar
constituent[14]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/wagon-http-3.1.0-shaded.jar
constituent[15]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-model-builder-3.5.4.jar
constituent[16]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/commons-io-2.5.jar
constituent[17]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/slf4j-api-1.7.25.jar
constituent[18]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/plexus-component-annotations-1.7.1.jar
constituent[19]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/plexus-cipher-1.7.jar
constituent[20]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-shared-utils-3.2.1.jar
constituent[21]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/guava-20.0.jar
constituent[22]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/wagon-file-3.1.0.jar
constituent[23]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/org.eclipse.sisu.plexus-0.3.3.jar
constituent[24]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-core-3.5.4.jar
constituent[25]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/javax.inject-1.jar
constituent[26]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-settings-3.5.4.jar
constituent[27]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/wagon-provider-api-3.1.0.jar
constituent[28]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-embedder-3.5.4.jar
constituent[29]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/org.eclipse.sisu.inject-0.3.3.jar
constituent[30]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-builder-support-3.5.4.jar
constituent[31]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-resolver-provider-3.5.4.jar
constituent[32]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-resolver-spi-1.1.1.jar
constituent[33]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/commons-lang3-3.5.jar
constituent[34]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-resolver-connector-basic-1.1.1.jar
constituent[35]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/aopalliance-1.0.jar
constituent[36]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-model-3.5.4.jar
constituent[37]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-resolver-api-1.1.1.jar
constituent[38]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/jansi-1.17.1.jar
constituent[39]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-resolver-transport-wagon-1.1.1.jar
constituent[40]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/plexus-sec-dispatcher-1.4.jar
constituent[41]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/maven-slf4j-provider-3.5.4.jar
constituent[42]: file:/opt/hostedtoolcache/maven/3.5.4/x64/lib/cdi-api-1.0.jar
---------------------------------------------------
Error: Exception in thread "main" java.lang.UnsupportedClassVersionError: com/fasterxml/jackson/databind/ObjectMapper : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:808)
```

Where is it coming from? com/fasterxml/jackson/databind/ObjectMapper

**Comment by suztomo** [comment]
After commenting out "nexus-staging-maven-plugin", now the error is on os-maven-plugin:

```
[ERROR] Internal error: com.google.inject.ProvisionException: Unable to provision, see the following errors:
[ERROR] 
[ERROR] 1) Error in custom provider, java.lang.TypeNotPresentException: Type kr.motd.maven.os.DetectExtension not present
[ERROR]   at ClassRealm[extension>kr.motd.maven:os-maven-plugin:1.7.1, parent: sun.misc.Launcher$AppClassLoader@60437dcb] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)
[ERROR]   while locating org.apache.maven.AbstractMavenLifecycleParticipant annotated with @com.google.inject.name.Named(value=detect-os)
[ERROR] 
[ERROR] 1 error: kr/motd/maven/os/DetectExtension : Unsupported major.minor version 52.0
[ERROR] -> [Help 1]
```

File: .github/workflows/ci-java7.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        # google-http-client-jackson2 and google-http-client-appengine do not work with Java 7
+        # any more because of Jackson and appengine library are compiled for Java 8.
+        # dailymotion-simple-cmdline-sample and google-http-client-assembly depend on
+        # google-http-client-jackson2 
+        mvn --batch-mode --show-version -ntp test \
+            --projects '!google-http-client-jackson2,!google-http-client-appengine,!samples/dailymotion-simple-cmdline-sample,!google-http-client-assembly' \
+            -Dclirr.skip=true -Denforcer.skip=true -Dmaven.javadoc.skip=true \
+            -Dgcloud.download.skip=true -T 1C \
+            -Dproject.surefire.version=2.22.2 \
+            -Djvm=${JAVA7_HOME}/bin/java
```
Comment (suztomo) [inline .github/workflows/ci-java7.yaml:62]: Memo: This is the core of this enhancement. https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#jvm

**Comment by suztomo** [comment]
Guava's warning about Java 7 tells that the test is indeed running on Java 7:

image

https://github.com/googleapis/google-http-java-client/actions/runs/4927699269/jobs/8805047743?pr=1847

File: .github/workflows/ci-java7.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        # This value is used in "-Djvm=" later
+        echo "JAVA7_HOME=${JAVA_HOME}" >> $GITHUB_ENV
+    - uses: actions/setup-java@v3
+      with:
+        java-version: 17
+        distribution: temurin
+    - name: Set up Maven
+      uses: stCarolas/setup-maven@v4.5
+      with:
+        maven-version: 3.8.8
```
Comment (lqiu96) [inline .github/workflows/ci-java7.yaml:44]: I believe this is the last version of maven that supports Java 7, right? I don't think anyone is going to manually update this, but would it helpful to add a blurb as to why this set to `3.8.8` ?

**Review by lqiu96 — APPROVED** [review]
LGTM

File: .github/workflows/ci-java7.yaml (hunk 2 above)
Comment (suztomo) [reply .github/workflows/ci-java7.yaml:44]: Actually this Maven does not need to work for Java 7. In fact this check does not run Maven via Java 7. (Maven calls Java 7 when running surefire unit tests)

**Pull request merged**